packages/gatsby-plugin-page-creator/src/gatsby-node.js

```diff
@@ -15,14 +16,38 @@ const validatePath = require(`./validate-path`)
 // underscored. Then create url w/ our path algorithm *unless* user
 // takes control of that page component in gatsby-node.
 exports.createPagesStatefully = async (
-  { store, boundActionCreators },
-  options,
+  { store, boundActionCreators, reporter },
+  pluginOptions,
   doneCb
 ) => {
   const { createPage, deletePage } = boundActionCreators
   const program = store.getState().program
   const exts = program.extensions.map(e => `${e.slice(1)}`).join(`,`)
-  const pagesDirectory = systemPath.posix.join(program.directory, `/src/pages`)
+
+  if (!(pluginOptions && pluginOptions.path)) {
+    reporter.panic(
+      `
+      "path" is a required option for gatsby-plugin-page-creator
+
+      See docs here - https://www.gatsbyjs.org/packages/gatsby-plugin-page-creator/
+      `
+    )
+  }
+
+  // Validate that the path exists.
+  if (!fs.existsSync(pluginOptions.path)) {
```

Comment:
One downside to this is that if someone doesn't want to have a src directory or src/pages directory then this will error. Perhaps the core instance of this plugin could say it is optional so if src/pages doesn't exist, then it just does nothing.

Reply:
How should this be implemented? I can't see I understand fully?